onnxruntime/test/python/quantization/test_save_activations.py

```diff
@@ -0,0 +1,168 @@
+    )
+    graph.initializer.add().CopyFrom(w1)
+    graph.initializer.add().CopyFrom(b1)
+    graph.initializer.add().CopyFrom(w3)
+    graph.initializer.add().CopyFrom(b3)
+    graph.initializer.add().CopyFrom(w5)
+    graph.initializer.add().CopyFrom(b5)
+    model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 13)])
+    onnx.save(model, test_model_path)
+
+
+class TestDataReader(CalibrationDataReader):
+    """Random Data Input Generator"""
+
+    def __init__(self):
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.count = 2
+        self.input_data_list = []
+        for _ in range(self.count):
+            self.input_data_list.append(np.random.normal(0, 0.33, [1, 3, 1, 3]).astype(np.float32))
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            input_name = "input"
+            self.enum_data_dicts = iter([{input_name: input_data} for input_data in self.input_data_list])
+        return next(self.enum_data_dicts, None)
+
+    def rewind(self):
+        self.preprocess_flag = True
+
+
+class TestSaveActivations(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_model_dir = tempfile.TemporaryDirectory(prefix="test_save_activations.")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._tmp_model_dir.cleanup()
+
+    def test_saved_tensors_match_internal_tensors(self):
+        test_model_path = str(Path(self._tmp_model_dir.name) / "augmented_model.onnx")
+        construct_test_model1(test_model_path)
+        data_reader = TestDataReader()
+
+        aug_model = augment_model_save_tensors(test_model_path)
+        augmented_model_path = str(Path(self._tmp_model_dir.name).joinpath("augmented_test_model_1.onnx"))
+
+        onnx.save(
+            aug_model,
+            augmented_model_path,
+            save_as_external_data=False,
+        )
+
+        tensor_dict = run_collect_activations(augmented_model_path, data_reader)
+
+        # run original model and compare the tensors
+        sess_options = onnxruntime.SessionOptions()
+        sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL
+        infer_session = onnxruntime.InferenceSession(
+            test_model_path,
+            sess_options=sess_options,
+            providers=["CPUExecutionProvider"],
+        )
+
+        data_reader.rewind()
+        oracle_outputs = []
+        for input_d in data_reader:
+            oracle_outputs.append(infer_session.run(None, input_d))
+
+        output_dict = {}
+        output_info = infer_session.get_outputs()
+        for batch in oracle_outputs:
+            for output, output_data in zip(output_info, batch):
+                output_dict.setdefault(output.name, []).append(output_data)
+
+        for output_name, model_outputs in output_dict.items():
+            test_outputs = tensor_dict.get(output_name)
```

Comment:
test_outputs can be None if output_name does not exist in tensor_dict.

If output_name always exists in the tensor_dict, I would do `tensor_dict[output_name]`